### PR TITLE
TransferAgreementsController uses unitofwork

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs [hunk 1]
```diff
@@ -27,11 +27,10 @@ namespace API.Transfer.Api.Controllers;
 [ApiVersion(ApiVersions.Version20240103)]
 [Route("api/transfer/transfer-agreements")]
 public class TransferAgreementsController(
-    ITransferAgreementRepository transferAgreementRepository,
     IProjectOriginWalletService projectOriginWalletService,
     IHttpContextAccessor httpContextAccessor,
     ITransferAgreementProposalRepository transferAgreementProposalRepository,
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs:32]: Shouldnt this also be deleted?

File: domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public UnitOfWork(ApplicationDbContext context)
+    {
+        this.context = context;
+    }
+
+    public ITransferAgreementRepository TransferAgreementRepo
+    {
+        get
+        {
+            return transferAgreementRepo ??= new TransferAgreementRepository(context);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs:33]: Could we use DI here?

File: domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+using System.Threading.Tasks;
+using API.Transfer.Api.Repository;
+using DataContext;
+using EnergyOrigin.ActivityLog.API;
+
+namespace API.UnitOfWork;
+
+public interface IUnitOfWork
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs:8]: Shouldnt there be a rollback if any of these goes wrong? 
Have you read? https://learn.microsoft.com/en-us/archive/msdn-magazine/2009/june/the-unit-of-work-pattern-and-persistence-ignorance

Why is this implementation of unitofwork necessary compared to plain old DI? :)

Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs:8]: Because here we can do cross repo transactions. We can't with DI

Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs:8]: I havn't implemented rollback. THought that could be done when needed but can do it upfront if you want?

File: domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs (hunk 2 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/UnitOfWork/IUnitOfWork.cs:33]: To what purpose?

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs (hunk 1 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs:32]: Yes. Needs to be enforced throughout the solution as well

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs [hunk 4]
```diff
@@ -55,7 +52,7 @@ public async Task<ActionResult> Create(CreateTransferAgreement request)
         }
 
         var proposal =
-            await transferAgreementProposalRepository.GetNonExpiredTransferAgreementProposalAsNoTracking(request.TransferAgreementProposalId);
+            await unitOfWork.TransferAgreementProposalRepo.GetNonExpiredTransferAgreementProposalAsNoTracking(request.TransferAgreementProposalId);
```
Comment (ckr123) [inline domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs:55]: Something with the indention gone wrong here?

Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs:55]: You mean the newline? Corrected

**Review by ckr123 — APPROVED** [review]

**Pull request merged**